wemake_python_styleguide/patches/baseline.py

```diff
@@ -78,13 +84,89 @@ def _handle_results(  # noqa: WPS210, WPS430
                 saved_reports.append(
                     (error_code, line_number, column, text, physical_line),
                 )
-            # --- patch end
-        return reported_results_count
+
+        if self._wps_baseline:
+            self._wps_baseline.write_file(self.options.baseline)
+
+        # We should exit successfully if created a baseline.
+        return 0 if self.options.create_baseline else reported_results_count
+        # --- patch end
 
     manager._handle_results = _handle_results  # noqa: WPS437
 
 
+def _patch_start(manager: Type[Manager]) -> None:
+    original_start = manager.start
+
+    def start(self, paths=None) -> None:  # noqa: WPS430
+        # --- patch start
+        self._wps_baseline = None
+        if self.options.create_baseline:
+            if paths is not None or self.arguments:
+                response = input("This will create a new baseline for only the given files. Continue? (y/n) ")
```

Comment:
Is this a development only `input`?

Reply:
No, is there some flake8 mechanism for getting user input?

The idea here is to double check the user's intentions if they've passed explicit files, as I expect in most cases that would be a mistake. Usually, you'd just run it across the entire project.

I added this on a hunch. But, a couple of hours later, this saved me from making a mistake.

Reply:
No, there's no such mechanism. Let's leave it there for now, but we cannot not keep it in the final solution.